### ISSUE TITLE
Instance.update is calling Model.create

### ISSUE THREAD

**Issue by ryanmcclure4** [sequelize/sequelize#9272]
## What are you doing?
I am calling _.update_ on a sequelize model instance and it is in turn attempting to _insert_ a new record with the provided fields. I cannot remember which version this started happening, but it seemed one day all of my _.update_ calls triggered an _insert_ query. Model.update, however works fine.

```js
const user = await User.findOne({ ... });

// This triggers a `INSERT INTO "users" ("field") VALUES("thing") ...`
user.update({ field: 'thing' });

// This updates the model just fine
User.update({ field: 'thing' }, { where: { id: 1 } });
```

## What do you expect to happen?
The instance model should be updated

## What is actually happening?
An INSERT query is called rather than an UPDATE query

**Comment by sushantdhiman** [comment]
"One day all of my .update calls triggered an insert query" not much to work with

This could happen when `isNewRecord: true` or model missing `id`(primary key). I need a proper SSCCE to reproduce and tag this as a bug

Please use Github Issue Tracker only for reporting bugs, requesting new features or discussions. Ask questions on [Stackoverflow sequelize.js tag](https://stackoverflow.com/questions/tagged/sequelize.js) or [Slack](sequelize.slack.com). While I would like to help answer any questions but it still take too much time.

If you are reporting a bug please isolate it as SSCCE, present it nicely and follow issue template. This will help maintainers and contributors understand your issue quickly.

If you are requesting a feature spend some time to properly present your use case with some examples, current and expected outcome.

( [What is SSCCE ?](http://sscce.org) You can find template for Sequelize [here](https://github.com/sequelize/sequelize/blob/master/sscce_template.js) )

A few good example of SSCCE

1) https://github.com/sequelize/sequelize/issues/8014#issue-245907597
2) https://github.com/sequelize/sequelize/issues/7318#issuecomment-284136530
3) https://github.com/sequelize/sequelize/issues/8749#issuecomment-354734396

**Comment by ryanmcclure4** [comment]
I figured this would be an issue others were experiencing due to a recent version change, however I realized it is because of a change I made to the caching layer in my app. I am serializing the instance data and caching that, and then calling Model.build(data) after fetching from the cache. I then call _update_ on this instance, so this must be causing the issue as _build_ doesn't include the id and sets `isNewRecord: true`.
